lib/system.nim

```diff
@@ -4170,7 +4170,15 @@ when declared(File):
   template `&=`*(f: File, x: typed) = write(f, x)
 
 template currentSourcePath*: string = instantiationInfo(-1, true).filename
-  ## returns the full file-system path of the current source
+  ## Returns the full file-system path of the current source.
+  ## This is the path of the Nim source file containing this code.
+  ## It is available at compile-time only.
```

Comment:
> It is available at compile-time only.

Ah, sorry; I didn't actually test this before, but this template actually works when the executable is being ran too and not just during compile-time, so just remove this sentence. The restrictions on compile-time execution are clearly stated in the manual, anyway.

That's why you should test anything that you document if you're not sure about whether it's correct :) And if you're still not sure, ask questions!

Reply:
But I tested it. 🤷‍♀️

Reply:
Hmm, I tested with this (with devel):

```nim
static:
  echo "compile time: " & currentSourcePath

echo "program execution: " & currentSourcePath
```